extractor/src/test/java/org/schabi/newpipe/extractor/services/media_ccc/MediaCCCRecentListExtractorTest.java

```diff
@@ -24,16 +28,18 @@ public static void setUpClass() throws Exception {
     }
 
     @Test
-    public void testStreamList() throws Exception {
+    void testStreamList() throws Exception {
         final List<StreamInfoItem> items = extractor.getInitialPage().getItems();
-        assertEquals(100, items.size());
-        for (final StreamInfoItem item: items) {
-            assertFalse(isNullOrEmpty(item.getName()));
-            assertTrue(item.getDuration() > 0);
-            // Disabled for now, because sometimes videos are uploaded, but their release date is in the future
-            // assertTrue(item.getUploadDate().offsetDateTime().isBefore(OffsetDateTime.now()));
-        }
-    }
+        assertFalse(items.isEmpty(), "No items returned");
 
+        assertAll(items.stream().flatMap(this::getAllConditionsForItem));
+    }
 
+    private Stream<Executable> getAllConditionsForItem(final StreamInfoItem item) {
+        return Stream.of(
+                () -> assertFalse(isNullOrEmpty(item.getName())),
+                () -> assertTrue(item.getDuration() > 0,
+                        "Duration[=" + item.getDuration() + "] of " + item + "is <= 0")
+        );
+    }
```

Comment:
Wouldn't a simple foreach have been better in this case? This seems highly convoluted. Also, use `assertGreater`.
```suggestion
        items.stream().forEach(item -> {
            assertFalse(isNullOrEmpty(item.getName()));
            assertGreater(0, item.getDuration());
        })
    }
```

Reply:
> Wouldn't a simple foreach have been better in this case? 

No.
→ https://stackoverflow.com/a/40800468

>  Also, use assertGreater.

👍

Reply:
> No.

Oh, didn't know that! That's great news :-)